ref/general/csec-tls.adoc

```diff
@@ -0,0 +1,108 @@
+. The client verifies the server's digital certificate and sends an encrypted random byte string that is used to generate the shared secret key. If the server requested the client's certificate, the client sends either the certificate or a `no certificate alert`. This alert can cause the handshake to fail if the certificate is configured as mandatory by the server. If a certificate is sent, the server verifies the client's certificate.
+. The client sends a `finished` message, which is encrypted with the shared secret key and indicates that the client side of the handshake is complete.
+. The server sends a `finished` message, which is encrypted with the shared secret key and indicates that the server side of the handshake is complete.
+. For the remainder of the session, client and server can exchange symmetrically encrypted messages by using the shared secret key.
+
+=== The TLS record
+
+Once the TLS handshake establishes authentication and encryption details, the TLS record protocol handles the transmission and verification of encrypted messages during the session.
+The TLS record reads the messages to be transmitted, breaks them into manageable chunks and encrypts them according to the terms agreed on during the TLS handshake.
+The TLS record also specifies a means to compute the message digest so that message integrity can be verified by both parties.
+On the receiving side, the record decrypts the messages and verifies their integrity by comparing the message digests.
+
+== TLS and Open Liberty
+
+You can enable support for TLS in Open Liberty by adding the `transportSecurity-1.0` feature to your `server.xml` file.
+You can specify the details of your TLS settings by adding the  `ssl`, `sslDefault`, and `keyStore` elements to your `server.mxl` file.
+For more information, see the link:/docs/ref/feature/#transportSecurity.html[Transport Security feature].
+
+In addition to adding the `transportSecurity-1.0` feature, the `keystore` element must be added to your `server.xml` file.
+The following example shows the minimum configuration to enable TLS.
+
+[source,java]
+----
+<feature>transportSecurity-1.0</feature>
+
+<keyStore id="defaultKeyStore" password="yourPassword" />
+----
+
+Open liberty creates a keystore password during profile creation and puts it in the server.env file that is in the server's home directory.
+If there is no keystore element for the `defaultKeyStore` file, this password is used to create a keystore file.
+This keystore file is then used as the `defaultKeyStore` file.
+Likewise, if a `defaultKeyStore` entry exists without a password in the `sever.xml` file when the server starts, the password from the server.env file is used to open the file.
+If you don't want to use the generated keystore password, remove the `keystore_password` entry from the server.env file.
+If a default keystore was already generated with the password from the `server.env` file, you might need to remove it.
```

Comment:
Not clear on whether "keystore" should be in backticks in this last instance.

Reply:
I hoping to make a distinction between a specific `keystore` file and the more generic reference to keystore as a type of file. This part is adapted from KC which doesn't use codeph for generic references. But if the distinction is not evident should probably rethink